tests/test_shape.py

```diff
@@ -0,0 +1,45 @@
+import re
+import pytest
+import numpy as np
+from pytest_cases import parametrize
+
+# from mrinufft.operators.interfaces.utils.utils import check_shape
+
+
+def check_shape(self_shape, image):
+    """Check if the image shape is compatible with the operator's init shape."""
+    image_shape_comparaison = image.shape[-len(self_shape) :]
+    if image_shape_comparaison != self_shape:
+        raise ValueError(
+            f"Image shape {image.shape[-len(self_shape):]} is not compatible "
+            f"with the operator shape {self_shape}"
+        )
+
+
+@parametrize(
+    "self_shape, image_shape",
+    [
+        ((256, 256), (16, 10, 256, 256)),
+        ((256, 256, 176), (16, 10, 256, 256, 176)),
+    ],
+)
+def test_check_shape_pass(self_shape, image_shape):
+    image = np.random.rand(*image_shape)
+    try:
+        check_shape(self_shape, image)
+    except ValueError:
+        pytest.fail("Unexpected ValueError raised")
+
+
+@parametrize(
+    "self_shape, image_shape",
+    [
+        ((256, 256), (16, 10, 256, 254)),
+        ((256, 256, 176), (16, 10, 256, 256)),
+    ],
+)
+def test_check_shape_fail(self_shape, image_shape):
+    image = np.random.rand(*image_shape)
+    expected_message = f"Image shape {image.shape[-len(self_shape):]} is not compatible with the operator shape {self_shape}"
+    with pytest.raises(ValueError, match=re.escape(expected_message)):
+        check_shape(self_shape, image)
```

Comment:
I dont think we need this test... It seems redundant...

Reply:
redundancy is not a bad things for tests. However, the testes `check_shape` must be the one in the utils module (otherwise this has no interest!)

Reply:
Oh come on, its literally same code in test xD

Reply:
@paquiteau i know,but strangely it cannot import the function , it's was the just a first draft to see if i need to change something , i'm working on that